Student_HG_Laravel/app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Validator;
+
+class AuthController extends Controller
+{
+    public function register(Request $request)
+    {
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'name'     => 'required|string|max:255',
+                'username' => 'required|string|max:10|unique:users',
+                'email'    => 'required|string|email|max:255|unique:users',
+                'password' => 'required|string|min:8',
+            ],
+            [
+                'name.required'     => 'Enter Your Name',
+                'username.required' => 'Enter Your Username',
+                'email.required'    => 'Enter Your Email',
+                'password.required' => 'Enter Your Password',
+            ]
+        );
+        if ($validator->fails()) {
+            return response()->json($validator->errors());
```

Comment:
Ini kurang statusnya mas. Nanti jangan lupa di update di deskripsi juga ya hasil testingnya 💪 

```suggestion
return response()->json(
     [
          'status' => 'failed',
          'message' => $validator->errors()->all()
     ]
);
```